uast/uast.go

```diff
@@ -262,87 +270,311 @@ type Any interface{}
+
+	// Tab is a set of whitespace and stylistic characters that appears at the beginning
+	// of each comment line, except the first one.
+	//
+	// Example:
+	//    /*
+	//     * some comment
+	//     */
+	//
+	//    the " *" before the comment text is considered a tab
+	Tab string `json:"Tab"`
 }
 
+// Group is a no-op UAST node that groups multiple nodes together.
+//
+// Drivers may use it when for grouping statements that are represented by a single statement
+// in the native AST.
+//
+// For example, a language may describe a way to define multiple variables in one statement.
+// This statement should be split into separate UAST nodes that become a children of a single Group.
+//
+// Groups should never convey any semantic meaning.
 type Group struct {
 	GenNode
+	// Nodes is a list of UAST nodes in a group.
 	Nodes []Any `json:"Nodes"`
 }
 
+// FunctionGroup is a special group node that joins multiple UAST nodes related to a function
+// declaration.
+//
+// FunctionGroup usually contains at least an Alias node that specifies the function name and
+// may contain additional nodes such as annotations and comments and docs related to it.
+//
+// See Function for more details about function declarations.
 type FunctionGroup Group
 
+// Block is a logical code block. It groups multiple statements and enforces a sequential execution
+// of these statements.
+//
+// When the Block should be used:
+// - for function bodies;
+// - when the statement defines a new scope;
 type Block struct {
 	GenNode
 	Statements []Any `json:"Statements"`
+
+	// TODO: block is logical and should have a reference to a corresponding scope; should be used in "if"s, etc
 	// Scope *Scope
 }
 
+// Alias provides a way to assign a permanent name to an entity, or give an alternative name.
+//
+// Aliases are immutable and the only way to redefine it is to shadow it in the child scope.
+//
+// What is considered an Alias:
+// - a name of a function in a function declaration;
+// - a name of a constant and its value;
+// - a name of a preprocessor macros and its substitution;
+// - variable declaration; // TODO: should point to some Variable node
+//
+// Not considered an Alias:
+// - value assignments to a variable, even if it defines a variable;
 type Alias struct {
 	GenNode
+	// Name assigned to an entity.
+	//
+	// TODO: define a different node to handle QualifiedIdentifier as a name
 	Name Identifier `json:"Name"`
-	Node Any        `json:"Node"`
+
+	// A UAST node to assign a name to.
+	Node Any `json:"Node"`
+
+	// TODO: should include a pointer to a scope where an alias is defined
 	// Target *Scope
 }
 
+// Import is a statement that can load other modules into the program or library.
+//
+// This is a declarative import statement. It's position in the UAST does not affect
+// the way and the time when the module is imported and the side-effects are executed
+// only once a package is initialized.
+//
+// Similar to import in Go, Java, C#.
```

Comment:
Rather than "similar to…" in these examples, I'd suggest wording it to more directly describe the relationship. Perhaps: "This describes imports in Go, Java, and C#, for example."